packages/adblocker-webextension/src/index.ts

```diff
@@ -401,11 +401,12 @@ export class WebExtensionBlocker extends FiltersEngine {
       this.config.enableHtmlFiltering === true &&
       browser.webRequest !== undefined &&
       browser.webRequest.filterResponseData !== undefined &&
-      request.isMainFrame() === true &&
       typeof TextDecoder !== 'undefined' &&
       typeof TextEncoder !== 'undefined'
     ) {
-      const htmlFilters = this.getHtmlFilters(request);
+      const htmlFilters = this.getHtmlFilters(request, {
+        selectors: request.isMainFrame() === true ? ['script'] : undefined,
```

Comment:
Main Frame should be filtered by both script (`has-text) and replace (`$replace) html selectors, while all other requests should only be handled by the replace.

@remusao this is one way of deciding which HTMLSelectors should be used for Filtering.

Alternatively, we can move this logic to `getHtmlFilters` but that would make it less elastic. Not sure what's better.